packages/expo-cli/src/commands/utils/validateDependenciesVersions.ts

```diff
@@ -27,34 +36,45 @@ export async function validateDependenciesVersionsAsync(
     return false;
   }
 
-  const bundledNativeModules = await JsonFile.readAsync(bundleNativeModulesPath);
   const bundledNativeModulesNames = Object.keys(bundledNativeModules);
   const projectDependencies = Object.keys(pkg.dependencies || []);
 
-  const modulesToCheck = intersection(bundledNativeModulesNames, projectDependencies);
-  const incorrectDeps = [];
-  for (const moduleName of modulesToCheck) {
-    const expectedRange = bundledNativeModules[moduleName];
-    const actualRange = pkg.dependencies[moduleName];
+  const packagesToCheck = intersection(bundledNativeModulesNames, projectDependencies);
```

Comment:
Since this code is being rewritten it'd be a good time to add a unit test, split it up, add a comment or two about what's going on, and maybe also drop the lodash function.